assets/src/edit-story/app/api/apiProvider.js

```diff
@@ -240,9 +240,30 @@ function APIProvider({ children }) {
     return apiFetch({ path });
   }, [statuses]);
 
-  const getAllUsers = useCallback(() => {
-    return apiFetch({ path: users });
-  }, [users]);
+  const getUsersByPage = useCallback(
+    (pagingNum = 1) => {
+      let apiPath = users;
+      const perPage = 100;
+      apiPath = addQueryArgs(apiPath, {
+        per_page: perPage,
+        page: pagingNum,
+      });
+      return apiFetch({ path: apiPath });
+    },
+    [users]
+  );
+
+  const getAllUsers = useCallback(
```

Comment:
We might be able to get the results quicking by using the `X-WP-TotalPages`, [see](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/media/mediaProvider.js#L73). Get the count of local pages and perform all the api requests at once. Did you try this when building this?

Reply:
No, I didn't tried it. I will try it now

Reply:
Yeah, now refactored and uses Promise.all once for all pages.